### ISSUE TITLE
allow_redirects default for 'head' not shown in help text

### ISSUE THREAD

**Issue by geokala** [psf/requests#4274]
When checking help(requests.request) to get the default parameters, the text is misleading if a head request is going to be used.

## Expected Result

The help text should indicate where the default differs OR the help text for .head should indicate that it overrides the default.

e.g. 
    :param allow_redirects: (optional) Boolean. Enable/disable GET/OPTIONS/POST/PUT/PATCH/DELETE/HEAD redirection. Defaults to ``False`` for head requests, ``True`` for all others.

## Actual Result

The 'head' help text displays only that it takes the args of a 'request' object, and request displays:
    :param allow_redirects: (optional) Boolean. Enable/disable GET/OPTIONS/POST/PUT/PATCH/DELETE/HEAD redirection. Defaults to ``True``.

## Reproduction Steps

```python
import requests
help(requests.head)
help(requests.request)
```

## System Information

    $ python -m requests.help

```
{
  "chardet": {
    "version": "3.0.4"
  }, 
  "cryptography": {
    "version": ""
  }, 
  "idna": {
    "version": "2.6"
  }, 
  "implementation": {
    "name": "CPython", 
    "version": "2.7.6"
  }, 
  "platform": {
    "release": "3.16.0-30-generic", 
    "system": "Linux"
  }, 
  "pyOpenSSL": {
    "openssl_version": "", 
    "version": null
  }, 
  "requests": {
    "version": "2.18.4"
  }, 
  "system_ssl": {
    "version": "1000106f"
  }, 
  "urllib3": {
    "version": "1.22"
  }, 
  "using_pyopenssl": false
}
```

**Comment by Lukasa** [comment]
HEAD overrides the default and should be updated.

**Comment by geokala** [comment]
I've created a PR for this here: https://github.com/requests/requests/pull/4277

I tried updating just in requests.head, but then the requests.request still has an incorrect docstring if it is only updated there so for a first pass I've updated in just the one place. I'm not sure if this is 'right' for requests, or if there are any similar examples (I had a quick skim through api.py but am possibly being oblivious this evening).

I also applied the same (almost) change to session.py since the same different default is present there.
